### DISCUSSION TITLE
Allow for undefined generated property

### DISCUSSION THREAD

**Discussion by robertteevans** [prisma/prisma#9347] [Q&A] [answered]
Suppose I have this schema:

```prisma
model Post {
  id Int @id @default(autoincrement())
  body String
  createdAt DateTime? @default(now())
}
```
The generated Typescript type will be defined as follows:

```ts
export type Post = {
  id: number
  body: string
  createdAt: string | null
}
```
The problem is that I want the signature to be "createdAt?: string | null". Is this possible?

**Comment by ryands17** [comment]
@robertteevans 👋 

Could you explain your use case for an `undefined` property in this one? I could suggest an alternative in this case.

**Comment by robertteevans** [comment, marked as answer]
Sure thing. I'm hoping to keep the code as terse as possible (and still type safe) and therefore I want to do something like this:

```ts
const posts: Post[] = postBodies.map(body => ({ body }));
await prisma.post.createMany({data: posts});
```

This results in the following error: 

`... missing the following properties from type 'Post': id, createdAt`

I'd prefer not to define the audit fields nor any auto generated field if I don't have to. Hope this helps.

**Reply by ryands17** [reply]
I see. Why not use Prisma's generated types for create inputs in this case:

```ts
import { Prisma } from '@prisma/client';

const posts: Prisma.PostCreateManyInput[] = []
```

**Reply by robertteevans** [reply]
Ah okay. I should have looked closer at the docs. Thanks for the quick clarification.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
